src/array_api_stubs/_draft/info.py

```diff
@@ -0,0 +1,168 @@
+__all__ = [
+    "__array_namespace_info__",
+    "capabilities",
+    "default_device",
+    "default_dtypes",
+    "devices",
+    "dtypes",
+]
+
+from ._types import (
+    Optional,
+    Union,
+    Tuple,
+    List,
+    device,
+    dtype,
+    DefaultDataTypes,
+    DataTypes,
+    Capabilities,
+    Info,
+)
+
+
+def __array_namespace_info__() -> Info:
+    """
+    Returns a namespace with Array API namespace inspection utilities.
+
+    Returns
+    -------
+    out: Info
+        An object containing Array API namespace inspection utilities.
+
+    Notes
+    -----
+
+    The returned object may be either a namespace or a class, so long as an Array API user can access inspection utilities as follows:
+
+    ::
+
+      info = xp.__array_namespace_info__()
+      info.capabilities()
+      info.devices()
+      info.dtypes()
+      info.default_dtypes()
+      # ...
+    """
+
+
+def capabilities() -> Capabilities:
+    """
+    Returns a dictionary of array library capabilities.
+
+    The dictionary should contain the following keys:
+
+    -   **boolean indexing**: boolean indicating whether an array library supports boolean indexing.
+    -   **data-dependent shapes**: boolean indicating whether an array library supports data-dependent output shapes.
```

Comment:
I probably missed the discussion on this, but what's the difference between these two things?

Reply:
Also it would be good to be specific here about what a `True` value for the capability must entail.

Reply:
Boolean indexing is related, but a library can support APIs which have data-dependent output shapes (e.g., `unique`), but, in principle, not boolean indexing, and vice versa.

Reply:
Q: @asmeurer is your question resolved?

Reply:
@asmeurer I included guidance about what a `True` value must entail. Does this satisfy your concern?

Reply:
Yes.